src/equations/lattice_boltzmann_2d.jl

```diff
@@ -0,0 +1,461 @@
+follows:
+
+```
+  6  2  5
+   ╲ │ ╱
+    ╲│╱
+  3──9──1
+    ╱│╲
+   ╱ │ ╲
+  7  4  8
+```
+
+The corresponding opposite directions are:
+* 1 ←→ 3
+* 2 ←→ 4
+* 3 ←→ 1
+* 4 ←→ 2
+* 5 ←→ 7
+* 6 ←→ 8
+* 7 ←→ 5
+* 8 ←→ 6
+* 9 ←→ 9
+
+The main sources for the base implementation were
+* Misun Min, Taehun Lee, **A spectral-element discontinuous Galerkin lattice Boltzmann method for
+  nearly incompressible flows**, J Comput Phys 230(1), 2011
+  [doi:10.1016/j.jcp.2010.09.024](https://doi.org/10.1016/j.jcp.2010.09.024)
+* Karsten Golly, **Anwendung der Lattice-Boltzmann Discontinuous Galerkin Spectral Element Method
+  (LB-DGSEM) auf laminare und turbulente nahezu inkompressible Strömungen im dreidimensionalen
+  Raum**, Master Thesis, University of Cologne, 2018.
+"""
+struct LatticeBoltzmannEquation2D{RealT<:Real, CollisionOp} <: AbstractLatticeBoltzmannEquation{2, 9}
+  c::RealT
+  c_s::RealT
+  rho0::RealT
+
+  Ma::RealT
+  u0::RealT
+
+  Re::RealT
+  L::RealT
+  nu::RealT
+
+  weights::SVector{9, RealT}
+  v_alpha1::SVector{9, RealT}
+  v_alpha2::SVector{9, RealT}
+
+  collision_op::CollisionOp
+end
+
+function LatticeBoltzmannEquation2D(; Ma, Re, collision_op=collision_bgk,
+                                    c=1, L=1, rho0=1, u0=nothing, nu=nothing)
+  # Sanity check that exactly one of Ma, u0 is not `nothing`
+  if isnothing(Ma) && isnothing(u0)
+    error("Mach number `Ma` and reference speed `u0` may not both be `nothing`")
+  elseif !isnothing(Ma) && !isnothing(u0)
+    error("Mach number `Ma` and reference speed `u0` may not both be set")
+  end
+
+  # Sanity check that exactly one of Re, nu is not `nothing`
+  if isnothing(Re) && isnothing(nu)
+    error("Reynolds number `Re` and visocsity `nu` may not both be `nothing`")
+  elseif !isnothing(Re) && !isnothing(nu)
+    error("Reynolds number `Re` and visocsity `nu` may not both be set")
+  end
+
+  # Calculate speed of sound
+  c_s = c / sqrt(3)
+
+  # Calculate missing quantities
+  if isnothing(Ma)
+    Ma = u0 / c_s
+  elseif isnothing(u0)
+    u0 = Ma * c_s
+  end
+  if isnothing(Re)
+    Re = u0 * L / nu
+  elseif isnothing(nu)
+    nu = u0 * L / Re
+  end
+
+  # Promote to common data type
+  Ma, Re, c, L, rho0, u0, nu = promote(Ma, Re, c, L, rho0, u0, nu)
+
+  # Source for weights and speeds: https://cims.nyu.edu/~billbao/report930.pdf
+  weights  = @SVector [1/9, 1/9, 1/9, 1/9, 1/36, 1/36, 1/36, 1/36, 4/9]
+  v_alpha1 = @SVector [ c,   0,  -c,   0,   c,   -c,   -c,    c,    0 ]
+  v_alpha2 = @SVector [ 0,   c,   0,  -c,   c,    c,   -c,   -c,    0 ]
+
+  LatticeBoltzmannEquation2D(c, c_s, rho0, Ma, u0, Re, L, nu,
+                             weights, v_alpha1, v_alpha2,
+                             collision_op)
+end
+
+
+get_name(::LatticeBoltzmannEquation2D) = "LatticeBoltzmannEquation2D"
+varnames_cons(equations::LatticeBoltzmannEquation2D) = @SVector ["pdf"*string(v) for v in eachvariable(equations)]
```

Comment:
```suggestion
varnames_cons(equations::LatticeBoltzmannEquation2D) = ntuple(v -> "pdf"*string(v), eachvariable(eqs))
```
That should fix the error. Sorry!